test/libsolidity/semanticTests/various/create_random.sol

```diff
@@ -0,0 +1,25 @@
+contract A {
+    uint256 public immutable createdOn = block.number;
+}
+
+contract C {
+    function addr() external returns (address) {
+        return address(this);
+    }
+
+    function test() external returns (address a1, address a2) {
+        bytes memory creation = type(A).creationCode;
+
+        assembly {
+            a1 := create(0, add(creation, 32), mload(creation))
+            a2 := create2(0, add(creation, 32), mload(creation), address())
+        }
+    }
+}
+
+// ====
+// compileViaYul: false
+// ----
+// addr() -> 0xc06afe3a8444fc0004668591e8306bfb9968e79e
```

Comment:
```
cast compute-address 0x1212121212121212121212121212120000000012 --nonce 1
Computed Address: 0xc06afE3A8444fc0004668591e8306bfB9968E79E
```
Should be really `nonce 0`.

Reply:
Don't remember how the testing framework works. I'm assuming that `contract A` is not deployed directly.

Reply:
Just read the code, there's a TODO for that: https://github.com/ethereum/solidity/pull/13533/files#diff-0b947a700f82b453dfef24ae4301f8e1f511428310cd15130bb90758fcde08c8R255

Reply:
There are two ways to do it:
1) check if the account has code
2) when creating an account set the nonce to 1